### PR TITLE
Port to Zig and package manager

### PR DESCRIPTION
**Pull request by LordMZTE**
This ports this to be a pure-zig module, getting rid of the libgamemode header.

Additionally, mach-gamemode has been updated to be a module usable with the package manager.

### CODE REVIEW

**Comment by emidoots** [comment]
I think according to Zig style guide `linux_impl` and `noop_impl` should be `LinuxImpl` and `NoopImpl`, right?

This looks like an awesome starting point either way, so I'm merging. I didn't test it, but if there are issues we can work through those as needed.

**Review by emidoots — APPROVED** [review]

File: .gitattributes [hunk 1]
```diff
@@ -1,2 +0,0 @@
-* text=auto eol=lf
-upstream/** linguist-vendored
```
Comment (emidoots) [inline .gitattributes:2]: We should keep this .gitattributes file, though. `* text=auto eol=lf` in specific ensures Windows contributors can't mess up with line endings so easily. The `upstream/**` part is harmless to have.

I'll still merge, though. I'll just add this back later.

**Review by emidoots — APPROVED** [review]

File: .gitattributes (hunk 1 above)
Comment (LordMZTE) [reply .gitattributes:2]: Sorry, I missed that first line!

**Comment by LordMZTE** [comment]
> I think according to Zig style guide `linux_impl` and `noop_impl` should be `LinuxImpl` and `NoopImpl`, right?
> 
> This looks like an awesome starting point either way, so I'm merging. I didn't test it, but if there are issues we can work through those as needed.

Actually, this is correct because structs in Zig also serve as modules, and since these impl structs/modules don't have any fields and aren't ever instanced, they should follow module naming convention.

**Pull request merged**